From 448b3610660813a584749f32058d92bd9c5b9d03 Mon Sep 17 00:00:00 2001
From: Thomas Latham <T.Latham@warwick.ac.uk>
Date: Tue, 20 Apr 2021 14:34:10 +0100
Subject: [PATCH] Fix compilation with Pythia 8.304

Summary: Pythia 8.304 switches to shared pointers (from raw) for ParticleDataEntry returns. Introduce version-dependent typedef to deal with this.

Test Plan: Compiled successfully with 8.304, 8.303, 8.244

Reviewers: jback, kreps

Reviewed By: kreps

Maniphest Tasks: T144

Differential Revision: https://phab.hepforge.org/D68
---
 R02-00-00/src/EvtGenExternal/EvtPythiaEngine.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/R02-00-00/src/EvtGenExternal/EvtPythiaEngine.cpp b/R02-00-00/src/EvtGenExternal/EvtPythiaEngine.cpp
index df38ec8..72aa46d 100644
--- a/R02-00-00/src/EvtGenExternal/EvtPythiaEngine.cpp
+++ b/R02-00-00/src/EvtGenExternal/EvtPythiaEngine.cpp
@@ -38,6 +38,12 @@
 #include <iostream>
 #include <sstream>
 
+#if PYTHIA_VERSION_INTEGER < 8304
+typedef Pythia8::ParticleDataEntry* ParticleDataEntryPtr;
+#else
+typedef Pythia8::ParticleDataEntryPtr ParticleDataEntryPtr;
+#endif
+
 using std::endl;
 
 EvtPythiaEngine::EvtPythiaEngine( std::string xmlDir, bool convertPhysCodes,
@@ -430,9 +436,9 @@ void EvtPythiaEngine::updateParticleLists()
         // or assignment will give it a different memory address and it will no longer refer to
         // the original particleData information from the generator pointer.
 
-        Pythia8::ParticleDataEntry* entry_generic =
+        ParticleDataEntryPtr entry_generic =
             _genericPythiaGen->particleData.particleDataEntryPtr( PDGCode );
-        Pythia8::ParticleDataEntry* entry_alias =
+        ParticleDataEntryPtr entry_alias =
             _aliasPythiaGen->particleData.particleDataEntryPtr( PDGCode );
 
         // Check that the PDG code is not zero/null and exclude other
-- 
2.30.2

